src/components/utils/Markdown.tsx

```diff
@@ -0,0 +1,165 @@
+
+// Required to display accurate font sizes for markdown elements.
+const StyledMarkdownWrapper = styled(Box)`
+  h1,
+  h2,
+  h3,
+  h4,
+  h5,
+  h6 {
+    font-size: inherit;
+    font-weight: 500;
+  }
+
+  blockquote {
+    margin: revert;
+  }
+
+  ol,
+  ul {
+    margin-left: 20px;
+  }
+  h1 {
+    font-size: 2em;
+  }
+
+  h2 {
+    font-size: 1.5em;
+  }
+
+  h3 {
+    font-size: 1.17em;
+  }
+
+  h4 {
+    font-size: 1em;
+  }
+
+  h5 {
+    font-size: 0.83em;
+  }
+
+  h6 {
+    font-size: 0.67em;
+  }
+
+  hr {
+    background-color: #000000;
+    height: 2px;
+    border: 0px;
+  }
+`;
+
+export interface MarkdownProps {
+  text: string;
+}
+
+const TitleBar = ({ language, code }: { language?: string; code: string }) => {
+  return (
+    <Row
+      mainAxisAlignment="flex-start"
+      crossAxisAlignment="stretch"
+      expand
+      justifyContent="space-between"
+      alignItems="center"
+      px="10px"
+      py="5px"
+      backgroundColor="#f5f5f5"
+      borderBottom="1px solid #eee"
+      borderRadius="6px"
+    >
+      <Box>{language || "plaintext"}</Box>
+      <CopyCodeButton code={code} />
+    </Row>
+  );
+};
+
+const CopyCodeButton = ({ code }: { code: string }) => {
+  const [copied, setCopied] = useState(false);
+
+  const handleCopyButtonClick = async (e: MouseEvent) => {
+    e.stopPropagation(); // Prevent this from triggering edit mode in the parent.
+
+    if (await copySnippetToClipboard(code)) setCopied(true);
+  };
+
+  useEffect(() => {
+    if (copied) {
+      const timer = setTimeout(() => setCopied(false), 2000);
+      return () => clearTimeout(timer);
+    }
+  }, [copied]);
+
+  return (
+    <Button
+      // TODO fix this
+      // @ts-ignore
+      onClick={handleCopyButtonClick}
+      size="xs"
+      variant="ghost"
+      px="5px"
+      _hover={{ background: "#EEEEEE" }}
+    >
+      <CopyIcon boxSize={4} mr={1} /> {copied ? "Copied!" : "Copy Code"}
+    </Button>
+  );
+};
+
+export function Markdown({ text }: { text: string }) {
+  const markdown = useMemo(
+    () => (
+      <StyledMarkdownWrapper>
+        <ReactMarkdown
+          remarkPlugins={[remarkGfm]} // supports github flavored markdown.
+          components={{
+            code({ node, inline, className, children, ...props }) {
+              const match = /language-(\w+)/.exec(className || "");
+              const code = String(children);
+              return !inline ? (
+                <Box
+                  padding="0"
+                  borderRadius="0.25rem"
+                  overflow="hidden"
+                  css={{
+                    "> div": { margin: "0 !important" },
+                    ".fa": { fontStyle: "normal !important" },
+                  }}
+                  {...props}
+                >
+                  <TitleBar language={match?.[1]} code={code} />
+                  <SyntaxHighlighter
+                    children={code}
+                    style={coy as any}
```

Comment:
hmm why do we have to do as any here?

Reply:
Good question. This was one of the things I wanted to clean up.

Was a weird type mismatch that I couldn't get to the bottom of, but found the solution [here](https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/479#issuecomment-1267772279):

![Screenshot 2023-04-03 at 3 43 45 PM](https://user-images.githubusercontent.com/25194960/229623329-5b31965d-db19-415b-bce9-71030cc5de84.png)

Reply:
🔥 

complete side note: github syntax highlighting looks rly good and is probably super battle tested? do we know what they use? 

this seems like a start — tho i guess a lot of it is ruby https://github.com/github/markup